recipes/_configure_secret.rb

```diff
@@ -0,0 +1,37 @@
+# coding: UTF-8
+#
+# Cookbook Name:: cerner_splunk
+# Recipe:: _configure_secret
+#
+# Configures the system splunk.secret file
+
+secret_hash = CernerSplunk::DataBag.load node['splunk']['config']['secrets'], pick_context: CernerSplunk.keys(node), type: :vault
+
+# We can't keep windows from quasi-starting on package install and we don't
+# yet allow changing the secret file once it exists so don't set it for windows.
+if !secret_hash || platform_family?('windows')
+  Chef::Log.info 'Splunk Secrets either not configured for this node or secrets not currently supported for windows.'
+  return
+end
+
+secret = secret_hash['splunk.secret']
+if secret
+  secret_path = ::File.join(node['splunk']['home'], 'etc', 'auth', 'splunk.secret')
+
+  ruby_block 'Check splunk.secret file' do
+    block do
+      CernerSplunk.validate_secret_file(secret_path, secret)
+    end
+  end
+
+  template 'splunk.secret' do
+    path secret_path
+    source 'splunk.secret.erb'
```

Comment:
Instead of using a template that just writes the contents to a file, we could use just a file resource with the contents as an argument just like we did with the admin password: https://github.com/cerner/cerner_splunk/blob/1.14.1/recipes/_generate_password.rb#L38